Month3/PixelGameEngine/AsteroidGame/AsteroidMain.cpp

```diff
@@ -0,0 +1,463 @@
+
+
+			}
+
+		}
+	}
+
+	void AppendNewAsteroids()
+	{
+		for (const auto& NewAsteroid : NewAsteroids)
+		{
+			VecAsteroid.push_back(NewAsteroid);
+		}
+		NewAsteroids.clear();
+	}
+
+	void UpdateAndDrawAsteroids(float ElapsedTime)
+	{
+		for (auto& AsteroidObject : VecAsteroid)
+		{
+			AsteroidObject.PosX += AsteroidObject.DisplacementX * ElapsedTime;
+			AsteroidObject.PosY += AsteroidObject.DisplacementY * ElapsedTime;
+			WrapCoordinates(AsteroidObject.PosX, AsteroidObject.PosY, AsteroidObject.PosX, AsteroidObject.PosY);
+			DrawWireFrameModel(ModelAsteroids, AsteroidObject.PosX, AsteroidObject.PosY, AsteroidObject.Angle, AsteroidObject.Size);
+
+		}
+	}
+
+	//Bullets
+	void ShotBullet(float ElapsedTime)
+	{
+		if (GetKey(olc::Key::SPACE).bReleased)
+		{
+			Bullets.push_back({ Player.PosX,Player.PosY,50.0f * sinf(Player.Angle),-50.0f * cosf(Player.Angle),0,0 });
+		}
+	}
+
+	void UpdateAndDrawBullets(float ElapsedTime)
+	{
+		for (auto& Bullet : Bullets)
+		{
+			Bullet.PosX += Bullet.DisplacementX * ElapsedTime;
+			Bullet.PosY += Bullet.DisplacementY * ElapsedTime;
+			WrapCoordinates(Bullet.PosX, Bullet.PosY, Bullet.PosX, Bullet.PosY);
+			Draw(Bullet.PosX, Bullet.PosY);
+
+
+			for (auto& Asteroid : VecAsteroid)
+			{
+				if (Collition(Asteroid.PosX, Asteroid.PosY, Asteroid.Size, Bullet.PosX, Bullet.PosY))
+				{
+					//Asteroid hit
+					Bullet.PosX = -200;
+					if (Asteroid.Size > 4)
+					{
+						float Angle1 = ((float)rand() / (float)RAND_MAX) * 6.283185f;
+						float Angle2 = ((float)rand() / (float)RAND_MAX) * 6.283185f;
+						NewAsteroids.push_back({ Asteroid.PosX, Asteroid.PosY,10.0f * sinf(Angle1),10.0f * cosf(Angle1),(int)Asteroid.Size >> 1, 0.0f });
+						NewAsteroids.push_back({ Asteroid.PosX, Asteroid.PosY,10.0f * sinf(Angle2),10.0f * cosf(Angle2),(int)Asteroid.Size >> 1, 0.0f });
+					}
+					Asteroid.PosX = -100;
+					Score += 100;
+				}
+			}
+		}
+	}
+
+
+
+	void DestroyBullets()
+	{
+		if (Bullets.size() > 0)
+		{
+			auto Destroy =
+				remove_if(Bullets.begin(), Bullets.end(), [&](SpaceObject BulletDestroy)
+					{
+						return (BulletDestroy.PosX < 1 || BulletDestroy.PosY < 1 || BulletDestroy.PosX >= ScreenWidth() || BulletDestroy.PosY >= ScreenHeight());
+					});
+			if (Destroy != Bullets.end())
+			{
+				Bullets.erase(Destroy);
+			}
+		}
+	}
+
+	void DestroyAsteroids()
+	{
+		if (VecAsteroid.size() > 0)
+		{
+			auto Destroy =
+				remove_if(VecAsteroid.begin(), VecAsteroid.end(), [&](SpaceObject AsteroidDestroy)
+					{
+						return (AsteroidDestroy.PosX < 0);
+					});
+			if (Destroy != VecAsteroid.end())
+			{
+				VecAsteroid.erase(Destroy);
+			}
+		}
+	}
+
+	bool Collition(float AsteroidCenterX, float AsteroidCenterY, float Radius, float PosX, float PosY)
+	{
+		return sqrt(((PosX - AsteroidCenterX) * (PosX - AsteroidCenterX)) + ((PosY - AsteroidCenterY) * (PosY - AsteroidCenterY))) < Radius;
+	}
+
+	void ShipCollition()
```

Comment:
this is not a good name for this method, can try it with a bool function and name like IsCollided